### PR TITLE
Suspended contract improvement

### PR DESCRIPTION
**Pull request by magiccookie**
#2628

### CODE REVIEW

File: apps/ehealth/lib/ehealth/employees/employees.ex [hunk 1]
```diff
@@ -152,7 +152,8 @@ defmodule EHealth.Employees do
 
   def create_or_update_employee(%Request{data: %{"employee_id" => employee_id} = employee_request}, req_headers) do
     employee = get_by_id!(employee_id)
-    party = employee |> Map.get(:party, %{}) |> Map.get(:id) |> Parties.get_by_id!()
+    party_id = employee |> Map.get(:party, %{}) |> Map.get(:id)
```
Comment (AlexKovalevych) [inline apps/ehealth/lib/ehealth/employees/employees.ex]: you already have party in employee.party field

File: apps/ehealth/lib/ehealth/employees/employees.ex [hunk 2]
```diff
@@ -176,43 +177,39 @@ defmodule EHealth.Employees do
...
       {:ok, employee}
     end
   end
 
-  defp suspend_legal_entity_owner_contracts(
-         %{"employee_type" => @type_owner, "legal_entity_id" => legal_entity_id},
-         headers
-       ) do
-    contracts = owner_contracts_to_suspend(legal_entity_id, headers)
+  defp suspend_all_party_contracts(party_id, req_headers) do
```
Comment (AlexKovalevych) [inline apps/ehealth/lib/ehealth/employees/employees.ex]: i would leave just `headers`

File: apps/ehealth/lib/ehealth/employees/employees.ex [hunk 3]
```diff
@@ -176,43 +177,39 @@ defmodule EHealth.Employees do
...
+        contractor_owner_id: owner.id,
+        status: Contract.status(:verified),
+        is_suspended: false
+      }
+
+      {
+        :ok,
+        %Page{entries: contracts},
+        _
+      } = Contracts.list(get_contracts_params, nil, req_headers)
```
Comment (AlexKovalevych) [inline apps/ehealth/lib/ehealth/employees/employees.ex]: it supports multiple ids separated with comma

File: apps/ehealth/lib/ehealth/employees/employees.ex (hunk 2 above)
Comment (magiccookie) [reply apps/ehealth/lib/ehealth/employees/employees.ex]: ok

File: apps/ehealth/lib/ehealth/employees/employees.ex [hunk 4]
```diff
@@ -150,69 +150,66 @@ defmodule EHealth.Employees do
     |> PRMRepo.all()
   end
 
-  def create_or_update_employee(%Request{data: %{"employee_id" => employee_id} = employee_request}, req_headers) do
+  def create_or_update_employee(%Request{data: %{"employee_id" => employee_id} = employee_request}, headers) do
     employee = get_by_id!(employee_id)
-    party = employee |> Map.get(:party, %{}) |> Map.get(:id) |> Parties.get_by_id!()
+    party_id = employee |> Map.get(:party, %{}) |> Map.get(:id)
+    party = party_id |> Parties.get_by_id!()
```
Comment (AlexKovalevych) [inline apps/ehealth/lib/ehealth/employees/employees.ex]: `party = Parties.get_by_id!(employee.party.id)`

Comment (magiccookie) [reply apps/ehealth/lib/ehealth/employees/employees.ex]: +

File: apps/ehealth/lib/ehealth/employees/employees.ex (hunk 3 above)
Comment (magiccookie) [reply apps/ehealth/lib/ehealth/employees/employees.ex]: we intend get contract list by `contract_owner_id` not by `ids`

**Pull request merged**